viewer/src/public/migration/Cognite3DModel.ts

```diff
@@ -423,11 +426,17 @@ export class Cognite3DModel extends THREE.Object3D implements CogniteModelBase {
     b: number,
     applyToChildren = false
   ): Promise<number> {
-    const treeIndices = await this.determineTreeIndices(treeIndex, applyToChildren);
     const color: [number, number, number] = [r, g, b];
-    await treeIndices.forEach(idx => this.nodeColors.set(idx, color));
-    this.cadNode.requestNodeUpdate(treeIndices);
-    return treeIndices.count;
+    if (applyToChildren) {
```

Comment:
This code block is repeated several times (almost identical), and should probably be refactored into a function that is used everywhere.

Reply:
I've tried de-duplicating the code but turns out the lambda I had to introduce when doing this reduces performance by 20-30%. After a discussion we decided to keep it like it is and add comments about why.